.github/workflows/pypi-publish.yml

```diff
@@ -10,10 +10,12 @@ on:
 jobs:
   build-n-publish:
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write # for trusted publishing
```

Comment:
I thought I mentioned this at some point that when using OIDC you shouldn't build in the same job due to privilege elevation risks… This should be split into two jobs.